pkg/server/cloud/aws/instances.go

```diff
@@ -441,6 +450,38 @@ func (e *AwsEC2) retrieveOrAllocateHost(node *api.Node) (*string, error) {
 	return allocateOutput.HostIds[0], nil
 }
 
+func (e *AwsEC2) isHostAvailable(hostId *string) bool {
+	describeOutput, err := e.client.DescribeHosts(&ec2.DescribeHostsInput{
+		HostIds:    []*string{hostId},
```

Comment:
The `aws` package has a function for create a slice of string pointers, IIRC `aws.StringSlice()`.